src/regex.rs

```diff
@@ -139,3 +160,30 @@ fn is_trivial_regex(s: &regex_syntax::Expr) -> Option<&'static str> {
         _ => None,
     }
 }
+
+struct RegexVisitor<'v, 't: 'v> {
+    cx: &'v LateContext<'v, 't>,
+    spans: HashSet<Span>,
+}
+
+impl<'v, 't: 'v> Visitor<'v> for RegexVisitor<'v, 't> {
+    fn visit_block(&mut self, block: &'v Block) {
+        if_let_chain!{[
+            let Some(ref expr) = block.expr,
+            match_type(self.cx, self.cx.tcx.expr_ty(expr), &["regex", "re", "Regex"]),
```

Comment:
I think `expr_ty` is more expensive since it's a hashmap lookup over a Vec index :smile: . Then again, that table is probably in cache.

Meh, land it.